Frontend_Web_Development_React_Angular_Vue/React/Week 1/All_About_npm.md

```diff
@@ -0,0 +1,144 @@
+
+To run different versions of node.js on your system without any hinderance you need to download the **node version manager**
+<p>&nbsp;</p>
+
+## **NVM Installation on Windows** 
+
+To install or update nvm,you should run the [install script](https://github.com/nvm-sh/nvm#install--update-script) on your terminal.
+
+If you face any problem with the installation you can check out the complete procedure [here](https://github.com/coreybutler/nvm-windows)
+
+<p>&nbsp;</p>
+
+To verify the installation use this command
+```
+command -v nvm
+```
+
+For installing node.js using nvm use this command
+```
+$ nvm install node 
+```
+For verifying the version of node use this command
+```
+$ node --version
+```
+<p>&nbsp;</p>
+
+## **NVM Installation on OS X or Linux**
+To install or update nvm,you should run the [install script](https://github.com/nvm-sh/nvm#install--update-script) on your terminal.
+
+If you face any problem with the installation you can check out [this](https://github.com/nvm-sh/nvm#troubleshooting-on-linux).
+
+<p>&nbsp;</p>
+
+To verify the installation use this command
+```
+command -v nvm
+```
+
+For installing node.js using nvm use this command
+```
+$ nvm install node 
+```
+For verifying the version of node use this command
+```
+$ node --version
+```
+<p>&nbsp;</p>
+
+# Using Node Installer for downloading Node.js
+
+## **For Windows** 
+Download the windows installer from the [Node.js](https://nodejs.org/en/download/current/) website.
+Download the LTS(Long-Term Support) version.
+
+<p>
+<img src="assets/../../../assets/LTS_Installer.png"  width="80%">
+</p>
+
+Run the .msi file that you downloaded in previous step.
+Accept the license agreement and and accept the default installation settings.
+
+
+## **For macOs**
+Download the Mac installer from the [Node.js](https://nodejs.org/en/download/current/) website.
+<p>
+<img src="assests/../../../assets/LTS_MacOs.png"  width="80%">
+</p>
+
+Run the .pkg file that you downloaded in previous step.
+Accept the license agreement and and accept the default installation settings.
+<p>&nbsp;</p>
+
+# npm CLI
+
+There are a number of [npm CLI](https://docs.npmjs.com/cli/v7/commands) commands that are used by developers all around the world.
+
+<p>&nbsp;</p>
+
+## **Initialising npm from terminal**
+
+
+Create a folder using the mkdir command.
+</br>
+Create a JavaScript file inside this folder using the touch command.
+
+<p><img src="assets/../../../assets/How_To_Use_NPM1.png"></p>
+
+Now use this command for initialising npm 
+```
+npm init
+```
+After this you will get a number of entries to fill and you can fill them according to your preference. All this will lead to the creation of a file with name `package` in your main folder. Once you'll start installing external packages they'll also show up in this package file.
+
+[Here](https://www.npmjs.com/) you can search your desired packages. 
```

Comment:
- You can replace this link by `On [npm](https://www.npmjs.com/) you can search your desired packages.`